File: Skinet/API/Helpers/Pagination.cs
﻿namespace API.Helpers
{
    public class Pagination<T> where T : class
    {
        public Pagination(int pageIndex, int pageSize, int count, IReadOnlyList<T> items)
        {
            PageIndex = pageIndex;
            PageSize = pageSize;
            Count = count;
            Items = items;
        }

        public int PageIndex { get; set; }
        public int PageSize { get; set; }
        public int Count { get; set; } = 0;
        public IReadOnlyList<T> Items { get; set; }
    }
}

File: client/src/app/shared/models/pagination.ts
import {IProduct} from './product';

export interface IPagination {
    pageIndex: number;
    pageSize: number;
    count: number;
    items: IProduct[];
}

export class Pagination implements IPagination {
    pageIndex: number = 0;
    pageSize: number = 0;
    count: number = 0;
    items: IProduct[] = [];
}
